### PR TITLE
enable deepMerge for AsyncStorage.mergeItem 

### PR DESCRIPTION
**Pull request by xiaohanzhang**
**Apply same merge logic as ReactNative**

- [x] handle AsyncStorage.mergeItem with lodash.merge
- [x] includes tests

### CODE REVIEW

File: src/apis/AsyncStorage/__tests__/index-test.js [hunk 1]
```diff
@@ -1,5 +1,66 @@
 /* eslint-env mocha */
+import _ from 'lodash';
```
Comment (necolas) [inline src/apis/AsyncStorage/__tests__/index-test.js]: Please import the specific module you need. We're probably fine with a `typeof` check here though

File: src/apis/AsyncStorage/__tests__/index-test.js [hunk 2]
```diff
@@ -1,5 +1,66 @@
...
+      cb(null, ...args);
+    }
+  };
+  _waterfall();
+};
 
 suite('apis/AsyncStorage', () => {
-  test.skip('NO TEST COVERAGE', () => {});
+  suite('mergeLocalStorageItem', () => {
+    test('should have same behavior as react-navtive', (done) => {
```
Comment (necolas) [inline src/apis/AsyncStorage/__tests__/index-test.js]: `react-navtive` -> `react-native`

File: src/apis/AsyncStorage/index.js [hunk 3]
```diff
@@ -3,12 +3,13 @@
  * Copyright (c) 2015-present, Facebook, Inc.
  * All rights reserved.
  */
+import _ from 'lodash';
```
Comment (necolas) [inline src/apis/AsyncStorage/index.js]: `import merge from 'lodash/merge';`

**Comment by xiaohanzhang** [comment]
Hi, Is this ready for merge?

**Comment by necolas** [comment]
Thanks. Applied in ee7d3670626229f4d4c07bc730ab27f675faf394

**Comment by necolas** [comment]
Shipped in 0.0.47

**Pull request closed without merging**